File: dynamic/5/sol.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <assert.h>
#include <time.h>

char* longestPalindrome(char* s);

void runTest(char* s, char* expected, int testCase) {
    printf("Test case %d:\n", testCase);
    printf("Input: \"%s\"\n", s);
    printf("Expected: \"%s\"\n", expected);
    
    char* result = longestPalindrome(s);
    printf("Got: \"%s\"\n", result);
    
    // For cases with multiple valid answers, check length
    if (strcmp(result, expected) != 0) {
        assert(strlen(result) == strlen(expected));
        printf("Note: Multiple valid answers possible\n");
    }
    
    free(result);
    printf("Test case %d passed!\n\n", testCase);
}

void runPerformanceTest() {
    printf("Running performance test...\n");
    
    // Create a large palindromic string
    char* largeString = (char*)malloc(10001);
    memset(largeString, 'a', 10000);
    largeString[10000] = '\0';
    
    clock_t start = clock();
    
    for (int i = 0; i < 100; i++) {
        char* result = longestPalindrome(largeString);
        free(result);
    }
    
    clock_t end = clock();
    double time_spent = (double)(end - start) / CLOCKS_PER_SEC;
    
    free(largeString);
    printf("Performance test completed in: %f seconds\n\n", time_spent);
}

int main() {
    // Run performance test
    runPerformanceTest();
    
    // Test case 1: Example from problem statement
    runTest("babad", "bab", 1);
    
    // Test case 2: Second example
    runTest("cbbd", "bb", 2);
    
    // Test case 3: Single character
    runTest("a", "a", 3);
    
    // Test case 4: All same characters
    runTest("aaaa", "aaaa", 4);
    
    // Test case 5: Long palindrome
    runTest("xababababa", "ababababa", 5);
    
    // Test case 6: Empty string
    runTest("", "", 6);
    
    // Test case 7: Two characters
    runTest("ac", "a", 7);
    
    printf("All test cases passed!\n");
    return 0;
}

File: dynamic/5/sol.py
from my import Solution

if __name__ == "__main__":
    solution = Solution()
    
    # Test case 1: Example from problem statement
    s1 = "babad"
    result1 = solution.longestPalindrome(s1)
    print(f"Test case 1:")
    print(f"Input: {s1}")
    print(f"Expected: bab or aba")
    print(f"Got: {result1}")
    assert result1 in ["bab", "aba"], "Test case 1 failed"
    
    # Test case 2: Another example from problem statement
    s2 = "cbbd"
    result2 = solution.longestPalindrome(s2)
    print(f"\nTest case 2:")
    print(f"Input: {s2}")
    print(f"Expected: bb")
    print(f"Got: {result2}")
    assert result2 == "bb", "Test case 2 failed"
    
    # Test case 3: Single character
    s3 = "a"
    result3 = solution.longestPalindrome(s3)
    print(f"\nTest case 3:")
    print(f"Input: {s3}")
    print(f"Expected: a")
    print(f"Got: {result3}")
    assert result3 == "a", "Test case 3 failed"
    
    # Test case 4: All same characters
    s4 = "aaaa"
    result4 = solution.longestPalindrome(s4)
    print(f"\nTest case 4:")
    print(f"Input: {s4}")
    print(f"Expected: aaaa")
    print(f"Got: {result4}")
    assert result4 == "aaaa", "Test case 4 failed"
    
    # Test case 5: No palindrome longer than 1
    s5 = "abcd"
    result5 = solution.longestPalindrome(s5)
    print(f"\nTest case 5:")
    print(f"Input: {s5}")
    print(f"Expected: a or b or c or d")
    print(f"Got: {result5}")
    assert len(result5) == 1, "Test case 5 failed"
    
    print("\nAll test cases passed!")